clients/lsp-go.el

```diff
@@ -257,6 +257,52 @@ $GOPATH/pkg/mod along with the value of
   :group 'lsp-go
   :package-version '(lsp-mode "7.1"))
 
+(defcustom lsp-go-analyses (make-hash-table)
```

Comment:
the default is not good - hash-table is not alist. You should either change the default or the type of the defcustom.

Reply:
Sorry. Pushed a commit to change the default to `nil`. I also changed the default for `lsp-go-env`, as it had been `(make-hash-table)` also.

Tangentially, I'm not sure how to set a member of this setting to `false`. `'((unreachable . t))` seems to work to set `unreachable` to `true`, but `'((unreachable . nil))` or `'((unreachable . ()))` results in a `parsing setting "analyses": invalid type bool for map key "unreachable"` error, which appears to be coming from here: https://github.com/golang/tools/blob/5ab822f631a3f7b79c55b1e1dd18978ed2fcc312/internal/lsp/source/options.go#L1077. I'm not sure how to see the JSON that's actually being sent to `gopls`, but that error suggests that it's not `false`.

Reply:
use :json-false